aleph/views/bookmarks_api.py

```diff
@@ -1,51 +1,88 @@
 import logging
 from flask import Blueprint, request
-from aleph.views.util import jsonify, require
+
+from aleph.core import db
+from aleph.search import DatabaseQueryResult
+from aleph.views.util import jsonify, require, parse_request
+from aleph.views.serializers import BookmarkSerializer
+from aleph.model import Bookmark
 
 
 log = logging.getLogger(__name__)
 blueprint = Blueprint("bookmarks_api", __name__)
 
 
+@blueprint.route("/api/2/bookmarks", methods=["GET"])
+def index():
+    """Get a list of bookmarks created by the current user
+    ---
+    get:
+      summary: Get bookmarks
+      tags: [Bookmarks]
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json
+              schema:
+                type: object
+    """
+    require(request.authz.logged_in)
+
+    query = Bookmark.query.filter_by(role_id=request.authz.id)
+    result = DatabaseQueryResult(request, query)
+    return BookmarkSerializer.jsonify_result(result)
+
+
 @blueprint.route("/api/2/bookmarks", methods=["POST"])
 def create():
```

Comment:
This endpoint should handle the request gracefully when the/a given entity is already bookmarked and simply ignore the entity. Alternatively, if there is a more semantic way to express this in HTTP, it would also be fine to return a non-2xx status code. In that case, we’d handle this explicitly on the frontend.